cmake/test/gtest.cmake

```diff
@@ -1,6 +1,9 @@
 _generate_function_if_testing_is_disabled(
   "catkin_add_gtest"
   "catkin_add_gmock")
+_generate_function_if_no_cxx_language(
+  "catkin_add_gtest"
+  "catkin_add_gmock")
```

Comment:
This needs to include other public functions from this file, e.g. `catkin_add_executable_with_gtest`.

The same is probably needed above in `_generate_function_if_testing_is_disabled`.